keyvi/src/cpp/dictionary/dictionary_merger.h

```diff
@@ -87,105 +91,124 @@ final {
 
     private:
         EntryIteratorPtr entry_iterator_ptr_;
-        int priority_;
+        size_t segmentIndex_;
     };
 
- public:
-  DictionaryMerger(size_t memory_limit = 1073741824,
-                   const merger_param_t& params = merger_param_t()):
-                     dicts_to_merge_(), memory_limit_(memory_limit), params_(params){
-    if (params_.count(TEMPORARY_PATH_KEY) == 0) {
-      params_[TEMPORARY_PATH_KEY] = boost::filesystem::temp_directory_path().string();
+public:
+    DictionaryMerger(bool append_merge = false,
+                     size_t memory_limit = 1073741824, const merger_param_t& params = merger_param_t())
```

Comment:
it is more consistent if memory_limit is first, like for compiler. Why don't you put append_merge into the parameters? I would anyway not limit it to a bool. There is for sure more than 2 modes of merging.